pkg/keys/hashicorp_vault.go

```diff
@@ -72,6 +75,144 @@ func NewHCValueKeyID(keyID string) (*HCValueKeyID, error) {
 	}
 }
 
+type readKeyResponse struct {
+	Versions map[string]struct {
+		CreationTime time.Time `ms:"creation_time"`
+		PublicKeyPEM string    `ms:"public_key"`
+	} `ms:"keys"`
+}
+
+type vaultKeyVersion struct {
```

Comment:
compile time cast assert?

Reply:
i know it’s not necessary - but we normally do it.. so up to you

Reply:
I didn't originally add it because the return type from the method enforces it at compile time. Added.

Reply:
yeah - just trying to be internally consistent with what we've done elsewhere. Thanks!